test/common/http/http3/conn_pool_test.cc

```diff
@@ -122,9 +144,30 @@ TEST_F(Http3ConnPoolImplTest, FailWithSecretsBecomeEmpty) {
   EXPECT_EQ(static_cast<Http3ConnPoolImpl*>(pool.get())->instantiateActiveClient(), nullptr);
 }
 
-TEST_F(Http3ConnPoolImplTest, CreationWithBufferLimits) {
+class MockPoolConnectResultCallback : public PoolConnectResultCallback {
+public:
+  MOCK_METHOD(void, onConnectSucceeded, ());
+  MOCK_METHOD(void, onConnectFailedWithEarlyData, ());
+};
+
+TEST_F(Http3ConnPoolImplTest, CreationAndNewStream) {
   EXPECT_CALL(mockHost().cluster_, perConnectionBufferLimitBytes);
   initialize();
+
+  MockPoolConnectResultCallback connect_result_callback;
+  MockResponseDecoder decoder;
+  ConnPoolCallbacks callbacks;
+  pool_->setConnectResultCallback(connect_result_callback);
+
+  ConnectionPool::Cancellable* cancellable = pool_->newStream(decoder, callbacks);
+  EXPECT_NE(nullptr, cancellable);
+  std::list<Envoy::ConnectionPool::ActiveClientPtr>& clients =
+      Http3ConnPoolImplPeer::connectingClients(*pool_);
+  EXPECT_EQ(1u, clients.size());
```

Comment:
I think I would skip this check (and the addition of connectionClients) to the peer above. I'm not sure that it really adds to the test coverage.

Reply:
Oh, facepalm. Yes, you're right, of course.